Xamarin.Forms.Platform.Android/PopupManager.cs

```diff
@@ -112,9 +112,67 @@ void OnActionSheetRequested(Page sender, ActionSheetArguments arguments)
 				builder.Dispose();
 				//to match current functionality of renderer we set cancelable on outside
 				//and return null
+				if (arguments.FlowDirection == FlowDirection.MatchParent)
+				{
+					if (sender.FlowDirection == FlowDirection.MatchParent)
+						dialog.SetTitleFlowDirection(Device.FlowDirection);
+					else
+						dialog.SetTitleFlowDirection(sender.FlowDirection);
+				}
+				else
+				{
+					dialog.SetTitleFlowDirection(arguments.FlowDirection);
+				}
 				dialog.SetCanceledOnTouchOutside(true);
 				dialog.SetCancelEvent((o, e) => arguments.SetResult(null));
 				dialog.Show();
+
+				LayoutDirection layoutDirection = LayoutDirection.Ltr;
+				TextDirection textDirection = TextDirection.Ltr;
+
+				if (arguments.FlowDirection == FlowDirection.LeftToRight)
+				{
+					layoutDirection = LayoutDirection.Ltr;
+					textDirection = TextDirection.Ltr;
+				}
+				else if (arguments.FlowDirection == FlowDirection.RightToLeft)
+				{
+					layoutDirection = LayoutDirection.Rtl;
+					textDirection = TextDirection.Rtl;
+				}
+				else
+				{
+					if (sender.FlowDirection == FlowDirection.LeftToRight)
+					{
+						layoutDirection = LayoutDirection.Ltr;
+						textDirection = TextDirection.Ltr;
+					}
+					else if (sender.FlowDirection == FlowDirection.RightToLeft)
+					{
+						layoutDirection = LayoutDirection.Rtl;
+						textDirection = TextDirection.Rtl;
+					}
+					else
+					{
+						if (Device.FlowDirection == FlowDirection.LeftToRight)
+						{
+							layoutDirection = LayoutDirection.Ltr;
+							textDirection = TextDirection.Ltr;
+						}
+						else if (Device.FlowDirection == FlowDirection.RightToLeft)
+						{
+							layoutDirection = LayoutDirection.Rtl;
+							textDirection = TextDirection.Rtl;
+						}
+					}
```

Comment:
Same thing as the UWP implementation - I _think_ you can just look at the Page's `EffectiveFlowDirection` here instead of having to check for each possible setting of `FlowDirection`.